File: euler_4/euler_4.rb
def check_palindrome num
  stringed = num.to_s
  stringed.reverse == stringed
end

answer = 0

(100..999).each do |i|
  (100 .. 999).each do |j|
    num = i * j
    answer = num if check_palindrome(num) && answer < num
  end
end

p "The answer is #{answer}"

File: euler_4/euler_4.js
function reverseString(s) { return (s).split("").reverse().join("") };

function checkPalindrome(n) {
    var stringed = n + "";
    var reversed = reverseString(stringed);

    return stringed === reversed;
};

var answer = 0;
var num;

for (var i = 999; i > 99; --i) {
    for (var j = 999; j > 99; --j) {
        num = i * j;
        if (checkPalindrome(num) && answer < num) {
            answer = num;
        }
    }
}

console.log("The answer is: " + answer);
